controllers/user.js

```diff
@@ -0,0 +1,253 @@
+        res.json(list_of_all);
+    },
+
+    // Function for creating a new user
+    create: async (req, res) => {
+        // Check if the username already exists in the database
+        if (usernameValidator(req.body.username)) {
+            const name = await User.findOne({
+                where: {
+                    username: req.body.username,
+                },
+            });
+            // Check if the email already exists in the database
+            if (emailValidator(req.body.email)) {
+                const mail = await User.findOne({
+                    where: {
+                        email: req.body.email,
+                    },
+                });
+                // If both username and email are unique
+                if (name == null && mail == null) {
+                    // Check if the password is valid
+                    if (passwordValidator(req.body.password)) {
+                        const date = dateFormater(req.body.year, req.body.month, req.body.day);
+                        if (dateValidator(date)) {
+                            try {
+                                // Create the user and save it to the database
+                                await User.create({
+                                    username: req.body.username,
+                                    password: req.body.password,
+                                    email: req.body.email,
+                                    date_of_birth: date
+                                });
+                                res.send("user has been created: ");
+                            }
+                            catch {
+                                // If there was an error creating the user
+                                res.send("an error occurred while creating user");
+                            }
+                        }
+                        else
+                            res.send("not a valid date");
+                    }
+                    else
+                        // If the password is not valid
+                        res.send("not a valid password");
+                }
+                // If both username and email are not unique
+                else if ((name != null && mail != null)) {
+                    res.send("username and email already in use");
+                }
+                // If only username is not unique
+                else if (name != null)
+                    res.send("username already in use");
+                // If only email is not unique
+                else if (mail != null)
+                    res.send("email already in use");
+            }
+            else
+                res.send("not a valid email adress");
+        }
+        else
+            res.send("not a valid username");
+        // End the response
+        res.end();
+
+    },
+
+    // update_username function: updates the username for a given user
+    update_username: async (req, res) => {
```

Comment:
updates, have a lot of repetitive code. 

Update usually accepts the id, or username or the field which is used as a primary key in the db of the user to be updated, and the field to be updated. 

so. getUserById to find the user to be updated. 
update the field 
save.

Search in google how others implement it

Reply:
This way you will only use one update for all